sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -335,6 +353,39 @@ public static Task<WorkspaceStack> CreateOrSelectStackAsync(LocalProgramArgs arg
 
         private static readonly string[] SettingsExtensions = new string[] { ".yaml", ".yml", ".json" };
 
+        private async Task GetPulumiVersionAsync(CancellationToken cancellationToken = default)
+        {
+            var result = await this.RunCommandAsync(new[] { "version" }, cancellationToken).ConfigureAwait(false);
+            var versionString = result.StandardOutput.Trim();
+            if (versionString[0] == 'v') {
+                versionString = versionString.TrimStart('v');
+            }
+            var version = SemVersion.Parse(versionString);
+            if (version == null)
+            {
+                throw new ArgumentNullException("version");
+            }
+            LocalWorkspace.CheckVersionIsValid(LocalWorkspace._minimumVersion, version);
+            this.pulumiVersion = version;
+        }
+
+        public static void CheckVersionIsValid(SemVersion minVersion, SemVersion currentVersion)
+        {
+            var err = new InvalidOperationException($"Minimum version requirement failed. The minimum CLI version requirement is {minVersion}, your current CLI version is ${currentVersion}. Please update the Pulumi CLI.");
```

Comment:
It might be a preference but I would return `bool` and throw the exception at the caller side.

Reply:
Clarified the error messages a bit. I'm going to keep this as-is and just use it as a validator since it's not quite as straightforward as a true/false now.